query/src/lib/mutation.ts

```diff
@@ -0,0 +1,82 @@
+import { inject, InjectionToken } from '@angular/core';
+import { injectQueryClient } from './query-client';
+import {
+  DefaultError,
+  MutationObserver,
+  type MutationObserverOptions,
+  type MutationObserverResult,
+  notifyManager,
+} from '@tanstack/query-core';
+import { firstValueFrom, isObservable, Observable } from 'rxjs';
+import { toSignal } from '@angular/core/rxjs-interop';
+
+type CreateMutationOptions<
+  TData = unknown,
+  TError = DefaultError,
+  TVariables = void,
+  TContext = unknown
+> = Omit<
+  MutationObserverOptions<TData, TError, TVariables, TContext>,
+  'mutationFn'
+> & {
+  mutationFn: (variables: TVariables) => Promise<TData> | Observable<TData>;
+};
+
+class Mutation {
+  private instance = injectQueryClient();
+
+  use<
+    TData = unknown,
+    TError = DefaultError,
+    TVariables = unknown,
+    TContext = unknown
+  >(options: CreateMutationOptions<TData, TError, TVariables, TContext>) {
+    const mutationObserver = new MutationObserver<
+      TData,
+      TError,
+      TVariables,
+      TContext
+    >(this.instance, {
+      ...options,
+      mutationFn: (variables: TVariables): Promise<TData> => {
+        const result: Promise<TData> | Observable<TData> =
+          options.mutationFn(variables);
+        if (isObservable(result)) return firstValueFrom(result);
+        return result;
+      },
```

Comment:
One thing i think could be a problem, something that the old version also did, wrapping the `mutationFn` and converting the returned observable to a promise. By either using `firstValueFrom` or `new Promise((res, rej) => ...)` the http call would instantly resolve when using `reportProgress` in the Angulars Http Client, while this is okay for non `reportProgress` calls the flip side would just cancel it out.

Reply:
I don't think that we need to wrap it. Let's do the same as the query version